client/my-sites/plans-features-main/hooks/data-store/use-pricing-meta-for-grid-plans.ts

```diff
@@ -18,28 +22,88 @@ import type { IAppState } from 'calypso/state/types';
 interface Props {
 	planSlugs: PlanSlug[];
 	withoutProRatedCredits?: boolean;
+	storageAddOns?: ( AddOnMeta | null )[] | null;
+	currencyCode?: string | null;
+}
+
+/***
+ * The 2023 Pricing Grid handles prices differently than add-ons, so for the time being we need to
+ * do some conversion to calculate totals correctly.
+ *
+ * The 2023 pricing grid currently handles prices in some currencies in their smallest unit, like
+ * the Japanese Yen. For others, however, like USD, prices are not handled in their smallest unit,
+ * like the US dollar.
+ *
+ * For example, 2000 yen would be represented by the integer 2000 since a yen is the smallest unit.
+ * $20.00, however, would be represented as the integer 20, and not 2000. Since storage add-on
+ * prices are always handled in the smallest unit, we need to convert them, when appropriate, to the
+ * correct integer.
+ *
+ * We should expose a more friendly function from the format-currency package
+ */
+function convertPriceForSmallestUnit( price: number, currencyCode: string | null ) {
+	const currencyObject = getCurrencyObject( price, currencyCode || '', {
+		isSmallestUnit: true,
+	} );
+	return parseInt( currencyObject.integer.replace( /,/g, '' ) );
+}
+
+function getTotalPrices( planPrices: PlanPrices, addOnPrice = 0 ): PlanPrices {
+	const totalPrices = { ...planPrices };
+	let key: keyof PlanPrices;
+
+	for ( key in totalPrices ) {
+		const price = totalPrices[ key ];
+
+		if ( ! ( price === null ) ) {
```

Comment:
Out of curiosity, is there a special reason that this shouldn't be simply `price !== null`?

Reply:
Thanks for the patience! Will respond to this soon -- I got sidetracked with all of the currency work 😅

Reply:
Thanks for this suggestion -- I appreciate you pointing out the readability of these conditionals [aba3794](https://github.com/Automattic/wp-calypso/pull/81338/commits/aba379424440ed03263db52605f87347fa74fd51)